client/lib/domains/email-forwarding/index.js

```diff
@@ -19,12 +19,13 @@ import { type as domainTypes } from 'lib/domains/constants';
  */
 function getEligibleEmailForwardingDomain( selectedDomainName, domains = [] ) {
 	const eligibleDomains = getEmailForwardingSupportedDomains( domains );
+	let domainName;
 	if ( selectedDomainName ) {
 		eligibleDomains.forEach( function( domain ) {
-			return domain.name === selectedDomainName && selectedDomainName;
+			domainName = domain.name === selectedDomainName && selectedDomainName;
 		} );
 	}
-	return ( eligibleDomains.length && eligibleDomains[ 0 ].name ) || '';
+	return domainName || ( eligibleDomains.length && eligibleDomains[ 0 ].name ) || '';
```

Comment:
Random thoughts. If we're coming from the email tab on top, this is cool. But if we're inside a domain, and to to email forwarding, returning a different domain feels a but weird. What do you folks think?